lexis/data/Progress.h

```diff
@@ -0,0 +1,38 @@
+
+/* Copyright (c) 2016, Human Brain Project
+ *                     Stefan.Eilemann@epfl.ch
+ */
+
+#ifndef LEXIS_DATA_PROGRESS_H
+#define LEXIS_DATA_PROGRESS_H
+
+#include <lexis/api.h>
+#include <lexis/data/detail/progress.h> // base class
```

Comment:
ACK, but it's confusing due to the fact that "detail" is still used for PIMPL in multiple places. Also the other .fbs are used directly as a class, not put into a separate "hidden" namespace to be subclassed. So maybe the base Progress class should be in lexis::data, and this class should be renamed to DiscreteProgress or IntegerProgress, or it should use composition instead of inheritance. I see that this discussion is taking us beyond moving the code as you intended to do, so feel free to leave as is...